### DISCUSSION TITLE
Excessive buffer swaps while on vsync causes lags and performance issues

### DISCUSSION THREAD

**Discussion by AndGuz** [ppy/osu#21717] [Q&A] [answered]
### Type

Performance

### Bug description

If you set vsync the frame rate will be much lower than the screen frame rate and the performance counters shows a lot of buffer swapping happening, this doesn't happen when not using vsync. Im using Kde plasma on Arch Linux with X11 (happens too on wayland) 

Computer Specs:
-Intel N4000 Celeron Dual Core 1.1GHZ
-Intel UHD 600 Graphics
-Arch Linux
-Kde Plasma

### Screenshots or videos

This is with VSYNC
![image](https://user-images.githubusercontent.com/62262763/208470236-3fe6c191-3372-4c7c-b525-d664d2922c16.png)
This is without
![osu_2022-12-19_12-23-07](https://user-images.githubusercontent.com/62262763/208472268-868e383a-3783-40b3-88dc-4e1d11d070bb.jpg)

i get around 80-120 frames (sometimes peaks at 200) depending on gameplay or what objects are being drawn to screen.

### Version

2022.1205.0

### Logs

[network.log](https://github.com/ppy/osu/files/10260948/network.log)
[runtime.log](https://github.com/ppy/osu/files/10260949/runtime.log)
[performance.log](https://github.com/ppy/osu/files/10260950/performance.log)
[input.log](https://github.com/ppy/osu/files/10260951/input.log)
[database.log](https://github.com/ppy/osu/files/10260952/database.log)

**Comment by peppy** [comment, marked as answer]
This can happen due to some frame missing the vsync window, causing the GPU drive to vsync at half your expected frame (refresh) rate. It will improve alongside other performance work we do.

**Comment by AndGuz** [comment]
Maybe, but it continues happening even at lower resolutions, it is if the game isn't synchronizing properly to the display. Thanks for your fast reply. :+1:
